
From: Pete Zaitcev <zaitcev@redhat.com>

We at Red Hat shipped a larger number of arguments for quite some time, it
was required for installations on IBM mainframe (s390), which doesn't have
a good way to pass arguments.

There are a number of reasonable situations that go past the current limits
of 8.  One that comes to mind is when you want to perform a manual vnc
install on a headless machine using anaconda.  This requires passing in a
number of parameters to get anaconda past the initial (no-gui) loader
screens.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/init/main.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff -puN init/main.c~make-max_init_args-25 init/main.c
--- 25/init/main.c~make-max_init_args-25	2004-08-04 22:15:39.059095584 -0700
+++ 25-akpm/init/main.c	2004-08-04 22:15:39.065094672 -0700
@@ -105,8 +105,8 @@ EXPORT_SYMBOL(system_state);
 /*
  * Boot command-line arguments
  */
-#define MAX_INIT_ARGS 8
-#define MAX_INIT_ENVS 8
+#define MAX_INIT_ARGS 32
+#define MAX_INIT_ENVS 32
 
 extern void time_init(void);
 /* Default late time init is NULL. archs can override this later. */
_
